### PR TITLE
Naman/batch changes

### PR DESCRIPTION
**Pull request by thenamankumar**
- [x] Add calculator to notebooks page
- [x] Add batch changes page

## Batch Changes
CleanShot 2022-07-12 at 22 42 54@2x

## Notebooks
CleanShot 2022-07-12 at 22 43 12@2x

## Test plan

- Enable feature flag `admin-analytics-enabled`
- Visit `/site-admin/analytics/batch-changes`

### CODE REVIEW

File: internal/adminanalytics/batchchanges.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		COUNT(DISTINCT batch_changes.creator_id) AS unique_users,
+		COUNT(DISTINCT batch_changes.creator_id) AS registered_users
+	FROM
+		changesets
+		INNER JOIN batch_changes ON batch_changes.id = changesets.owned_by_batch_change_id
+	WHERE changesets.created_at %s
+	GROUP BY date
+`
+
+var changesetsCreatedSummaryQuery = `
```
Comment (eseliger) [inline internal/adminanalytics/batchchanges.go:29]: Do you want any changeset or only changesets that are published? if so, `publication_state` is the field you're looking for :)

File: internal/adminanalytics/batchchanges.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+		INNER JOIN changesets ON changesets.id = changeset_events.changeset_id
+		INNER JOIN batch_changes ON batch_changes.id = changesets.owned_by_batch_change_id
+	WHERE changeset_events.created_at %s AND changeset_events.kind IN (%s)
+`
+
+var mergeEventKinds = sqlf.Join([]*sqlf.Query{
+	sqlf.Sprintf("'github:merged'"),
+	sqlf.Sprintf("'bitbucketserver:merged'"),
+	sqlf.Sprintf("'gitlab:merged'"),
+}, ",")
```
Comment (eseliger) [inline internal/adminanalytics/batchchanges.go:89]: you also want `bitbucketcloud:pullrequest:fulfilled` :)

**Review by eseliger — APPROVED** [review]

File: client/web/src/site-admin/analytics/AnalyticsBatchChangesPage/index.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            },
+        ]
+
+        const calculatorProps = {
+            label: 'Changesets merged',
+            color: 'var(--cyan)',
+            value: changesetsMerged.summary.totalCount,
+            minPerItem: 15,
+            description:
+                'Batch Changes automates opening changesets across many repositories and codehosts. It also significantly reduces the time required to manage campaigns via tracking and management functions that are superior to custom solutions, spreadsheets and manually reaching out to developers.',
```
[the remark this answers is not in the document]
Comment (thenamankumar) [reply client/web/src/site-admin/analytics/AnalyticsBatchChangesPage/index.tsx]: updated

File: client/web/src/site-admin/analytics/components/TimeSavedCalculatorGroup.tsx [hunk 4]
```diff
@@ -173,3 +173,70 @@ export const TimeSavedCalculatorGroup: React.FunctionComponent<TimeSavedCalculat
...
+                        {formatNumber(hoursSaved)}
+                    </Text>
+                    <Text as="span" alignment="center">
+                        Hours saved
+                    </Text>
+                </div>
+            </div>
+            <div className="flex-1 d-flex flex-column m-0">
+                <Text as="span" weight="bold">
+                    About this statistics
```
Comment (rrhyne) [inline client/web/src/site-admin/analytics/components/TimeSavedCalculatorGroup.tsx:236]:
```suggestion
                    About this statistic
```

Comment (thenamankumar) [reply client/web/src/site-admin/analytics/components/TimeSavedCalculatorGroup.tsx:236]: Will be resolved in upcoming PR

**Comment by mrnugget** [comment]
@thenamankumar drive-by comment: we try to keep the git commit history on `main` relatively clean, so while PR titles don't matter too much (except to the author & reviewers?) I recommend doing a second pass over the commit message title/body when doing squash merge. (The commit that landed on `main` is called `Naman/batch changes` and makes it hard to figure out what's in it)

**Comment by thenamankumar** [comment]
> 

noted.

**Pull request merged**